[flaky test][exporter/loadbalancing] TestPeriodicallyResolveFailure data race

### Component(s)

exporter/loadbalancing

### Describe the issue you're reporting

https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/12927752205/job/36053593500#step:8:309

```
=== FAIL: . TestPeriodicallyResolveFailure (0.03s)
==================
WARNING: DATA RACE
Read at 0x00c00052e150 by goroutine 396:
  runtime.raceread()
      <autogenerated>:1 +0x24
  github.com/open-telemetry/opentelemetry-collector-contrib/exporter/loadbalancingexporter.(*dnsResolver).resolve()
      D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/exporter/loadbalancingexporter/resolver_dns.go:134 +0x88
  github.com/open-telemetry/opentelemetry-collector-contrib/exporter/loadbalancingexporter.(*dnsResolver).periodicallyResolve()
      D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/exporter/loadbalancingexporter/resolver_dns.go:121 +0x188
  github.com/open-telemetry/opentelemetry-collector-contrib/exporter/loadbalancingexporter.(*dnsResolver).start.gowrap1()
      D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/exporter/loadbalancingexporter/resolver_dns.go:96 +0x33

Previous write at 0x00c00052e150 by goroutine 395:
  runtime.racewrite()
      <autogenerated>:1 +0x24
  github.com/open-telemetry/opentelemetry-collector-contrib/exporter/loadbalancingexporter.(*dnsResolver).shutdown()
      D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/exporter/loadbalancingexporter/resolver_dns.go:110 +0xb5
  github.com/open-telemetry/opentelemetry-collector-contrib/exporter/loadbalancingexporter.TestPeriodicallyResolveFailure.func2()
      D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/exporter/loadbalancingexporter/resolver_dns_test.go:277 +0x44
  runtime.deferreturn()
      C:/hostedtoolcache/windows/go/1.22.8/x64/src/runtime/panic.go:602 +0x5d
  testing.tRunner()
      C:/hostedtoolcache/windows/go/1.22.8/x64/src/testing/testing.go:1689 +0x21e
  testing.(*T).Run.gowrap1()
      C:/hostedtoolcache/windows/go/1.22.8/x64/src/testing/testing.go:1742 +0x44
ERROR rerun aborted because previous run had a suspected panic and some test may not have run

Goroutine 396 (running) created at:
  github.com/open-telemetry/opentelemetry-collector-contrib/exporter/loadbalancingexporter.(*dnsResolver).start()
      D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/exporter/loadbalancingexporter/resolver_dns.go:96 +0x236
  github.com/open-telemetry/opentelemetry-collector-contrib/exporter/loadbalancingexporter.TestPeriodicallyResolveFailure()
      D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/exporter/loadbalancingexporter/resolver_dns_test.go:275 +0x9ea
  testing.tRunner()
      C:/hostedtoolcache/windows/go/1.22.8/x64/src/testing/testing.go:1689 +0x21e
  testing.(*T).Run.gowrap1()
      C:/hostedtoolcache/windows/go/1.22.8/x64/src/testing/testing.go:1742 +0x44

Goroutine 395 (running) created at:
  testing.(*T).Run()
      C:/hostedtoolcache/windows/go/1.22.8/x64/src/testing/testing.go:1742 +0x825
  testing.runTests.func1()
      C:/hostedtoolcache/windows/go/1.22.8/x64/src/testing/testing.go:2161 +0x85
  testing.tRunner()
      C:/hostedtoolcache/windows/go/1.22.8/x64/src/testing/testing.go:1689 +0x21e
  testing.runTests()
      C:/hostedtoolcache/windows/go/1.22.8/x64/src/testing/testing.go:2159 +0x8be
  testing.(*M).Run()
      C:/hostedtoolcache/windows/go/1.22.8/x64/src/testing/testing.go:2027 +0xf17
  go.uber.org/goleak.VerifyTestMain()
      C:/Users/runneradmin/go/pkg/mod/go.uber.org/goleak@v1.3.0/testmain.go:53 +0x64
  github.com/open-telemetry/opentelemetry-collector-contrib/exporter/loadbalancingexporter.TestMain()
      D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/exporter/loadbalancingexporter/generated_package_test.go:12 +0x2e4
  main.main()
      _testmain.go:251 +0x2d0
==================
panic: sync: WaitGroup is reused before previous Wait has returned [recovered]
	panic: sync: WaitGroup is reused before previous Wait has returned

goroutine 393 [running]:
testing.tRunner.func1.2({0x14[323](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/12927752205/job/36053593500#step:8:324)7460, 0x143bb0d60})
	C:/hostedtoolcache/windows/go/1.22.8/x64/src/testing/testing.go:1631 +0x3f7
testing.tRunner.func1()
	C:/hostedtoolcache/windows/go/1.22.8/x64/src/testing/testing.go:1634 +0x6b6
panic({0x143237460?, 0x143bb0d60?})
	C:/hostedtoolcache/windows/go/1.22.8/x64/src/runtime/panic.go:770 +0x132
sync.(*WaitGroup).Wait(0xc00052e148)
	C:/hostedtoolcache/windows/go/1.22.8/x64/src/sync/waitgroup.go:118 +0x105
github.com/open-telemetry/opentelemetry-collector-contrib/exporter/loadbalancingexporter.(*dnsResolver).shutdown(0xc00052e0c0, {0x0?, 0x0?})
	D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/exporter/loadbalancingexporter/resolver_dns.go:110 +0xb6
github.com/open-telemetry/opentelemetry-collector-contrib/exporter/loadbalancingexporter.TestPeriodicallyResolveFailure.func2()
	D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/exporter/loadbalancingexporter/resolver_dns_test.go:277 +0x45
github.com/open-telemetry/opentelemetry-collector-contrib/exporter/loadbalancingexporter.TestPeriodicallyResolveFailure(0xc000252000)
	D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/exporter/loadbalancingexporter/resolver_dns_test.go:286 +0xb4f
testing.tRunner(0xc000252000, 0x14387cf38)
	C:/hostedtoolcache/windows/go/1.22.8/x64/src/testing/testing.go:1689 +0x21f
created by testing.(*T).Run in goroutine 1
	C:/hostedtoolcache/windows/go/1.22.8/x64/src/testing/testing.go:1742 +0x826

=== FAIL: . TestSplitMetricsByResourceServiceName (unknown)

=== FAIL: . TestSplitMetricsByResourceServiceNameFailsIfMissingServiceNameAttribute (unknown)

=== FAIL: . TestSplitMetrics (unknown)

=== FAIL: . TestConsumeMetrics_SingleEndpoint (unknown)

=== FAIL: . TestConsumeMetrics_TripleEndpoint (unknown)
```

--- a/exporter/loadbalancingexporter/resolver_dns.go
+++ b/exporter/loadbalancingexporter/resolver_dns.go
@@ -93,6 +93,7 @@ func (r *dnsResolver) start(ctx context.Context) error {
 		r.logger.Warn("failed to resolve", zap.Error(err))
 	}
 
+	r.shutdownWg.Add(1)
 	go r.periodicallyResolve()
 
 	r.logger.Debug("DNS resolver started",
@@ -113,6 +114,7 @@ func (r *dnsResolver) shutdown(_ context.Context) error {
 
 func (r *dnsResolver) periodicallyResolve() {
 	ticker := time.NewTicker(r.resInterval)
+	defer r.shutdownWg.Done()
 
 	for {
 		select {
@@ -131,9 +133,6 @@ func (r *dnsResolver) periodicallyResolve() {
 }
 
 func (r *dnsResolver) resolve(ctx context.Context) ([]string, error) {
-	r.shutdownWg.Add(1)
-	defer r.shutdownWg.Done()
-
 	addrs, err := r.resolver.LookupIPAddr(ctx, r.hostname)
 	if err != nil {
 		r.telemetry.LoadbalancerNumResolutions.Add(ctx, 1, metric.WithAttributeSet(dnsResolverFailureAttrSet))

